test/functional/ui/float_spec.lua

```diff
@@ -10152,7 +10152,75 @@ describe('float window', function()
                                                   |
         ]])
       end
+      command('fclose!')
 
+      -- don't use winborder when reconfig a floating window
+      config.border = 'none'
+      local winid = api.nvim_open_win(buf, false, config)
+      config.border = nil
+      api.nvim_win_set_config(winid, config)
+      if multigrid then
+        screen:expect({
+          grid = [[
+          ## grid 1
+            [2:----------------------------------------]|*6
+            [3:----------------------------------------]|
+          ## grid 2
+            ^                                        |
+            {0:~                                       }|*5
+          ## grid 3
+                                                    |
+          ## grid 8
+            {1:none}|
+            {1: bor}|
+            {1:der }|
+            {2:~   }|
+          ]],
+          win_pos = {
+          [2] = {
+            height = 6,
+            startcol = 0,
+            startrow = 0,
+            width = 40,
+            win = 1000
+          }
+        },
+          float_pos = {
+          [8] = {1005, "NW", 1, 2, 2, true, 50};
+        },
+          win_viewport = {
+          [2] = {win = 1000, topline = 0, botline = 2, curline = 0, curcol = 0, linecount = 1, sum_scroll_delta = 0};
+          [8] = {win = 1005, topline = 0, botline = 2, curline = 0, curcol = 0, linecount = 1, sum_scroll_delta = 0};
+        },
+          win_viewport_margins = {
+          [2] = {
+            bottom = 0,
+            left = 0,
+            right = 0,
+            top = 0,
+            win = 1000
+          },
+          [8] = {
+            bottom = 0,
+            left = 0,
+            right = 0,
+            top = 0,
+            win = 1005
+          }
+        },
+        })
+      else
+        screen:expect([[
+          ^                                        |
+          {0:~                                       }|
+          {0:~ }{1:none}{0:                                  }|
+          {0:~ }{1: bor}{0:                                  }|
+          {0:~ }{1:der }{0:                                  }|
+          {0:~ }{2:~   }{0:                                  }|
+                                                  |
+        ]])
+      end
+      command('fclose!')
```

Comment:
This is way too much duplicate screen state. Can't you just use the previous floating window with `screen:expect_unchanged()`? (You need to assign the result of the previous `api.nvim_open_win` to `winid`)
```suggestion
      -- don't use winborder when reconfig a floating window
      config.border = nil
      api.nvim_win_set_config(winid, config)
      screen:expect_unchanged()
      command('fclose!')
```

Reply:
make sense